src/charm.py

```diff
@@ -21,25 +22,39 @@ class JujuControllerCharm(CharmBase):
 
     def __init__(self, *args):
         super().__init__(*args)
+
+        self.framework.observe(self.on.collect_unit_status, self._on_collect_status)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
-        self.framework.observe(self.on.start, self._on_start)
         self.framework.observe(
             self.on.dashboard_relation_joined, self._on_dashboard_relation_joined)
         self.framework.observe(
             self.on.website_relation_joined, self._on_website_relation_joined)
 
+        self._stored.set_default(db_bind_address='')
+        self._stored.set_default(last_bind_addresses=[])
+        self.framework.observe(
+            self.on.dbcluster_relation_changed, self._on_dbcluster_relation_changed)
+
         self.control_socket = controlsocket.Client(
             socket_path="/var/lib/juju/control.socket")
         self.framework.observe(
             self.on.metrics_endpoint_relation_created, self._on_metrics_endpoint_relation_created)
         self.framework.observe(
             self.on.metrics_endpoint_relation_broken, self._on_metrics_endpoint_relation_broken)
 
-    def _on_start(self, _):
-        self.unit.status = ActiveStatus()
+    def _on_collect_status(self, event: CollectStatusEvent):
+        if len(self._stored.last_bind_addresses) > 1:
```

Comment:
unless you're doing it on purpose for readability, you can equivalently:
```suggestion
        if self._stored.last_bind_addresses:
```

Reply:
That would be `> 0` though, right?

Reply:
Ah my bad, misread that. You're totally right.